### PR TITLE
Add basic SIMD support

### PR DESCRIPTION
**Pull request by DaseinPhaos**
- Add opt-in SIMD support for the crate. The feature requires crate
`simd` and specialization, thus can only be enabled under nightly.
The support can be turned on with `--features "use_simd"`.
Currently, the mainly supported types are limited to `Vector4<f32>`, `Matrix4<f32>` and `Quaternion<f32>`. Under the given benchmark certain operations were able
 to become up to 60% faster.

- Clean up some deadly tests. Also duplicate some new tests for SIMD.

- Clean up the build script for Travis.

#195

### CODE REVIEW

**Comment by brendanzab** [comment]
Oh wow, this is awesome!

File: benches/mat.rs [hunk 1]
```diff
@@ -59,3 +59,5 @@ bench_unop!(_bench_matrix4_invert, Matrix4<f32>, invert);
 bench_unop!(_bench_matrix2_transpose, Matrix2<f32>, transpose);
 bench_unop!(_bench_matrix3_transpose, Matrix3<f32>, transpose);
 bench_unop!(_bench_matrix4_transpose, Matrix4<f32>, transpose);
+
+bench_unop!(_bench_matrix4_determinant, Matrix4<f32>, determinant);
```
Comment (brendanzab) [inline benches/mat.rs]: Missing new line here

**Comment by phaazon** [comment]
**JUST YES!**

**Comment by DaseinPhaos** [comment]
@brendanzab Thanks for the review!

The source file have been updated accordingly.

**Comment by DaseinPhaos** [comment]
Seems like the current `nightly` version of `cargo` has some bugs causing the Travis build to fail. Downgrade to `nightly-2017-03-04` solved the problem. The corresponding build result can be found [here](https://travis-ci.org/DaseinPhaos/acgmath/jobs/210254818)

Specifically, I want to highlight the bench comparison:
 

```
                                                           no-SIMD          SIMD
test _bench_matrix4_add_m       ... bench:           9 ns/iter (+/- 0)     6 ns/iter (+/- 0)
test _bench_matrix4_determinant ... bench:          18 ns/iter (+/- 1)    16 ns/iter (+/- 0)
test _bench_matrix4_div_s       ... bench:          10 ns/iter (+/- 0)    10 ns/iter (+/- 0)
test _bench_matrix4_invert      ... bench:          74 ns/iter (+/- 1)    45 ns/iter (+/- 0)
test _bench_matrix4_mul_m       ... bench:          16 ns/iter (+/- 0)    16 ns/iter (+/- 0)
test _bench_matrix4_mul_s       ... bench:           8 ns/iter (+/- 0)     5 ns/iter (+/- 0)
test _bench_matrix4_mul_v       ... bench:           6 ns/iter (+/- 0)     5 ns/iter (+/- 0)
test _bench_matrix4_sub_m       ... bench:           9 ns/iter (+/- 0)     7 ns/iter (+/- 0)
test _bench_matrix4_transpose   ... bench:          17 ns/iter (+/- 1)    17 ns/iter (+/- 0)

test _bench_quat_add_q     ... bench:           3 ns/iter (+/- 2)     1 ns/iter (+/- 0)
test _bench_quat_conjugate ... bench:          10 ns/iter (+/- 0)    10 ns/iter (+/- 0)
test _bench_quat_div_s     ... bench:           4 ns/iter (+/- 0)     4 ns/iter (+/- 0)
test _bench_quat_invert    ... bench:           6 ns/iter (+/- 0)     4 ns/iter (+/- 0)
test _bench_quat_mul_q     ... bench:          10 ns/iter (+/- 0)     9 ns/iter (+/- 0)
test _bench_quat_mul_s     ... bench:           3 ns/iter (+/- 2)     3 ns/iter (+/- 0)
test _bench_quat_mul_v     ... bench:          13 ns/iter (+/- 0)    13 ns/iter (+/- 0)
test _bench_quat_normalize ... bench:           6 ns/iter (+/- 0)     5 ns/iter (+/- 0)
test _bench_quat_sub_q     ... bench:           3 ns/iter (+/- 0)     1 ns/iter (+/- 0)

test _bench_vector4_add_v     ... bench:           3 ns/iter (+/- 1)    1 ns/iter (+/- 0)
test _bench_vector4_div_s     ... bench:           4 ns/iter (+/- 0)    4 ns/iter (+/- 0)
test _bench_vector4_dot       ... bench:           5 ns/iter (+/- 0)    4 ns/iter (+/- 0)
test _bench_vector4_magnitude ... bench:           4 ns/iter (+/- 0)    4 ns/iter (+/- 0)
test _bench_vector4_mul_s     ... bench:           3 ns/iter (+/- 0)    3 ns/iter (+/- 0)
test _bench_vector4_normalize ... bench:           6 ns/iter (+/- 0)    5 ns/iter (+/- 0)
test _bench_vector4_rem_s     ... bench:          88 ns/iter (+/- 0)   89 ns/iter (+/- 19)
test _bench_vector4_sub_v     ... bench:           3 ns/iter (+/- 0)    1 ns/iter (+/- 0)
```

**Comment by brendanzab** [comment]
Reran travis - it passes!

**Pull request merged**